oci/archive/oci_dest.go

```diff
@@ -0,0 +1,128 @@
+}
+
+// SupportsSignatures returns an error (to be displayed to the user) if the destination certainly can't store signatures
+func (d *ociArchiveImageDestination) SupportsSignatures() error {
+	return d.ociImgDest.SupportsSignatures()
+}
+
+// ShouldCompressLayers returns true iff it is desirable to compress layer blobs written to this destination
+func (d *ociArchiveImageDestination) ShouldCompressLayers() bool {
+	return d.ociImgDest.ShouldCompressLayers()
+}
+
+// AcceptsForeignLayerURLs returns false iff foreign layers in manifest should be actually
+// uploaded to the image destination, true otherwise.
+func (d *ociArchiveImageDestination) AcceptsForeignLayerURLs() bool {
+	return d.ociImgDest.AcceptsForeignLayerURLs()
+}
+
+// MustMatchRuntimeOS returns true iff the destination can store only images targeted for the current runtime OS. False otherwise
+func (d *ociArchiveImageDestination) MustMatchRuntimeOS() bool {
+	return d.ociImgDest.MustMatchRuntimeOS()
+}
+
+// PutBlob writes contents of stream and returns data representing the result (with all data filled in).
+// inputInfo.Digest can be optionally provided if known; it is not mandatory for the implementation to verify it.
+// inputInfo.Size is the expected length of stream, if known.
+func (d *ociArchiveImageDestination) PutBlob(stream io.Reader, inputInfo types.BlobInfo) (types.BlobInfo, error) {
+	return d.ociImgDest.PutBlob(stream, inputInfo)
+}
+
+// HasBlob returns true iff the image destination already contains a blob with the matching digest which can be reapplied using ReapplyBlob
+func (d *ociArchiveImageDestination) HasBlob(info types.BlobInfo) (bool, int64, error) {
+	return d.ociImgDest.HasBlob(info)
+}
+
+func (d *ociArchiveImageDestination) ReapplyBlob(info types.BlobInfo) (types.BlobInfo, error) {
+	return d.ociImgDest.ReapplyBlob(info)
+}
+
+// PutManifest writes manifest to the destination.
+func (d *ociArchiveImageDestination) PutManifest(m []byte) error {
+	return d.ociImgDest.PutManifest(m)
+}
+
+func (d *ociArchiveImageDestination) PutSignatures(signatures [][]byte) error {
+	return d.ociImgDest.PutSignatures(signatures)
+}
+
+// Commit marks the process of storing the image as successful and asks for the image to be persisted
+// after the directory is made, it is tarred up into a file and the directory is deleted
+func (d *ociArchiveImageDestination) Commit() error {
+	if err := d.ociImgDest.Commit(); err != nil {
+		return errors.Wrapf(err, "error storing image %q", d.ref.tag)
+	}
+
+	// path of directory to tar up
+	_, src, _ := layout.GetOciReference(d.ref.ociRef)
+	// path to save tarred up file
+	dst := d.ref.resolvedDir
+	if err := tar(src, dst); err != nil {
+		return errors.Wrapf(err, "error tarring up directory %q", src)
+	}
+
+	return nil
+}
+
+// tar converts the direstory at src and saves it to dst
+func tar(src, dst string) error {
+	// input is a stream of bytes from the archive of the directory at path
+	input, err := archive.Tar(src, 0)
```

Comment:
The `archive.Uncompressed` typed constant. (The constant does the value 0, but it is typed and readable without checking the source code of the `archive` package)